pkg/updater/service/helper/main.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package main is a package that allows dd-updater
+// to execute a subset of priviledged commands
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"strings"
+	"syscall"
+	"unicode"
+)
+
+var (
+	installPath string
+	systemdPath = "/lib/systemd/system" // todo load it at build time from omnibus
+)
+
+func isValidChar(c rune) bool {
+	return unicode.IsLower(c) || c == '.' || c == '-' || c == ' '
+}
+
+func isValidString(s string) bool {
+	for _, c := range s {
+		if !isValidChar(c) {
+			return false
+		}
+	}
+	return true
+}
+
+func buildCommand(inputCommand string) (*exec.Cmd, error) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ Question:** ‏Can we test it independently while maintaining the "no import" axiom? I'm sure we'll test it in e2e though so this is no blocker.

Reply:
there are already some tests ehre:
- [unit tests](https://github.com/DataDog/datadog-agent/pull/23597/files#diff-7ab0e6279fc7f8319626afb20cdca8d806705e6731d8531f1716e09144d6f313R22-R27), to make them on darwin only some failure modes are tested
- [e2e](https://github.com/DataDog/datadog-agent/pull/23597/files#diff-8012ddaef67d823184533b23e8c729f433df372bfda32c22381a46e9645363ceR43-R54) tests the success path